charms/worker/k8s/src/token_distributor.py

```diff
@@ -260,13 +292,14 @@ def allocate_tokens(
             secret = relation.app.add_secret(content)
             secret.grant(relation, unit=unit)
             relation.data[self.charm.unit][secret_id] = secret.id or ""
+            app_databag[unit.name] = f"pending-{name}"
```

Comment:
# BIG CHANGE HERE

rather than JUST putting the nodename in the `app_databag`, let's put a state and the nodename

There are two states:
* `pending` meaning that we created a token, and it hasn't been used yet
* `joined` meaning we KNOW the other end used it